imap_processing/swapi/l2/swapi_l2.py

```diff
@@ -220,13 +220,22 @@ def swapi_l2(
     # -----------------------------------
     # Convert unpacked ESA_LVL5 values to hex to match the LUT table
     # value
-    esa_lvl5_hex = np.vectorize(lambda x: format(x, "X"))(l1_dataset["esa_lvl5"].values)
+    esa_lvl5_hex = np.vectorize(lambda x: format(x, "04X"))(
+        l1_dataset["esa_lvl5"].values
+    )
+
+    # TODO: We should use SPICE, but the dependencies aren't in sds-data-manager yet
+    # data_time = et_to_datetime64(ttj2000ns_to_et(l1_dataset["epoch"].data))
+    data_time = np.datetime64("2000-01-01T00:00:00", "ns") + np.asarray(
+        l1_dataset["epoch"].data
+    ).astype("timedelta64[ns]")
+
     esa_energy = solve_full_sweep_energy(
         esa_lvl5_hex,
         l1_dataset["sweep_table"].data,
         esa_table_df=esa_table_df,
         lut_notes_df=lut_notes_df,
-        data_time=np.array(l1_dataset["epoch"].data, dtype="datetime64[ns]"),
+        data_time=data_time,
```

Comment:
can you pass in ?`sci_start_time` instead? This is already in datetime format from l1?

Reply:
Can you make this minor change? I checked with Margaret and this will simply above changes and need to SPICE n L2.
```suggestion
        data_time=l1_dataset["sci_start_time"],
```

Reply:
OK, I pushed up something similar to this. But it did need a `.astype()` case because that variable is currently a string and you want it datetime64 in the receiving method. Take a look if that is what you were thinking.

Reply:
yep. That looks good to me!